src/use-target-changes/use-target-changes.ts

```diff
@@ -0,0 +1,158 @@
+import { StimulusUse, StimulusUseOptions } from '../stimulus_use'
+import { method } from '../support/index'
+import { TargetChangesController } from './target-changes-controller'
+
+export interface TargetChangesOptions extends StimulusUseOptions {
+
+}
+
+export class UseTargetChanges extends StimulusUse {
+  controller: TargetChangesController
+  observer: MutationObserver
+  targetElement: Node
+  targets: string[]
+  deprecatedTargets: string[]
+  options: TargetChangesOptions
+  targetSelector: string
+  deprecatedTargetSelector: string
+
+  constructor(controller: TargetChangesController, targets: string[], options: TargetChangesOptions = {}) {
+    super(controller, options)
+
+    this.targetElement = controller.element
+    this.targets = targets
+    this.deprecatedTargets = this.targets.map(target => `${controller.identifier}.${target}`)
+    this.controller = controller
+    this.deprecatedTargetSelector = `data-target`
+    this.targetSelector = `data-${this.controller.identifier}-target`
```

Comment:
for Stimulus 2 this should be ok 
https://github.com/hotwired/stimulus/blob/main/packages/%40stimulus/core/src/target_set.ts#L54
but I think it could change in the future, I remember seeing an issue on this.